### ISSUE TITLE
Avoid naming clashes with "get output" methods

### ISSUE THREAD

**Issue by danielrbradley** [pulumi/pulumi#8886]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

For a resource called `Xyz` we now have both the `getXyz(..)` and `getXyzOutputs(..)` invoke variation. However, if a module has a second resource called `XyzOutputs`  then we then also try to create `getXyzOutputs(..)` and `getXyzOutputsOutputs(..)`. There's now two functions called `getXyzOutputs(..)` which cause the SDK compillation to fail. 

Workarounds:

- Rename the invoke to avoid the clash (e.g. remove the trailing `s`) however this means the invoke doesn't match the resource. 
- Rename the whole resource, but this then means it doesn't match the original API (and might be a breaking change if already released).
- Disable generating invokes for one of the resources.
- Disable the `get...Outputs(..)` variation for the whole SDK.

Possible solutions:

- Optionally associate getters to resources so the names are scoped to the resource.
- Disable generation of `get...Outputs(..)` variation for single invokes.

### Affected area/feature

- Codegen for provider SDKs

**Comment by mikhailshilkov** [comment]
Did I understand you correctly that this is not blocking your progress and you applied one of the workarounds for now?

**Comment by danielrbradley** [comment]
Yes, using the workaround to not create the invoke for one of resource for now.
